webapp/components/create_comment.jsx

```diff
@@ -69,24 +73,85 @@ export default class CreateComment extends React.Component {
             submitting: false,
             ctrlSend: PreferenceStore.getBool(Constants.Preferences.CATEGORY_ADVANCED_SETTINGS, 'send_on_ctrl_enter'),
             showPostDeletedModal: false,
-            enableAddButton
+            enableAddButton,
+            showEmojiPicker: false,
+            emojiOffset: 0,
+            emojiPickerEnabled: Utils.isFeatureEnabled(Constants.PRE_RELEASE_FEATURES.EMOJI_PICKER_PREVIEW)
         };
 
         this.lastBlurAt = 0;
     }
 
+    closeEmoji(clickEvent) {
+        /*
+        if the user clicked something outside the component, except the RHS emojipicker icon
+        and the picker is open, then close it
+         */
+        if (clickEvent && clickEvent.srcElement &&
+            clickEvent.srcElement.className !== '' &&
+            clickEvent.srcElement.className.indexOf('emoji-rhs') === -1 &&
+            this.state.showEmojiPicker) {
+            this.setState({showEmojiPicker: !this.state.showEmojiPicker});
+        }
+    }
+
+    handleEmojiPickerClick() {
+        const threadHeight = $('#thread--root') ? $('#thread--root').outerHeight() : 0;
+        const messagesHeight = $('.post-right-comments-container') ? $('.post-right-comments-container').outerHeight() : 0;
```

Comment:
Well I don't know of any React-friendly way to go about it but can we use vanilla JS here instead of jQuery. Select with `document.querySelector()` for example to get the element. Not sure how easy it is to get `outerHeight` without jQuery. If it becomes a monumental task to do this or the below work without jQuery I guess I'll have to be OK with it but I'd prefer to leave that as a last resort.